File: src/main/java/it/tylframework/data/mongo/basics/Numerator.java
package it.tylframework.data.mongo.basics;

import it.tylframework.data.mongo.common.Footprint;
import it.tylframework.data.mongo.common.MlText;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.TypeAlias;

import java.util.ArrayList;

/**
 * Created by mp on 20/11/14.
 */
@TypeAlias("bas_Numerator")
@Data
@AllArgsConstructor
public class Numerator extends Footprint {
    String code;
    MlText name;
    MlText description;
    NumeratorType numerator_type;
    ArrayList<NumeratorFeeder> numerator_feeders = new ArrayList<NumeratorFeeder>();
}

File: src/main/java/it/tylframework/data/mongo/basics/Numerator.kt
package it.tylframework.data.mongo.basics

import it.tylframework.data.mongo.common.MlText
import it.tylframework.data.mongo.common.Footprint
import it.tylframework.data.mongo.common.Signature
import java.util.Date
import it.tylframework.data.mongo.config.TylContext
import org.springframework.data.annotation.TypeAlias

/**
 * Created by marco on 15/09/14.
 */
[TypeAlias("bas_Numerator")]
data class Numerator(
    var stdName: String,
    var name: MlText,
    var description: MlText,
    var numeratorType:NumeratorType,
    var numeratorFeeders: Array<NumeratorFeeder>)
    :Footprint(TylContext.currentUser, Date() ,TylContext.currentUser,  Date()) {
}